test/sampling.jl

```diff
@@ -13,6 +13,7 @@ x = [randi(10) for i = 1:n]
 @test isa(x, Vector{Int})
 @test extrema(x) == (1, 10)
 @test isapprox(proportions(x, 1:10), fill(0.1, 10), atol=5.0e-3)
+@test randi(MersenneTwister(1), 1000) == randi(MersenneTwister(1), 1000)
```

Comment:
It doesn't make a lot of sense to check whether the result is equal to itself. Any other way of testing this? What do the existing tests do for the existing methods? At least, checking that the result when passing the default RNG is equal what you would get without passing it would be better than this check. (Same below.)